packages/panna-sdk/src/core/helpers/http.ts

```diff
@@ -66,44 +66,51 @@ const AXIOS_ERR_CODE_TO_CUSTOM_ERR_MAP: Record<string, PannaHttpErr> = {
 };
 
 // Internal types
-type RequestParamsConfig = Omit<axios.AxiosRequestConfig, 'url'>;
+type RequestParamsConfig = Omit<AxiosRequestConfig, 'url'>;
 
 // Internal functions
-const isCacheableResponse = (response: axios.AxiosResponse): boolean =>
+export const isCacheableResponse = (response: AxiosResponse): boolean =>
   response.status === HttpStatusCode.Ok;
 
-const requestWithRetries = async (
+export const requestWithRetries = async (
   url: string,
-  requestParams: RequestParamsConfig,
+  requestParams: RequestParamsConfig = {},
   numRetries: number = DEFAULT_RETRIES,
   retryDelayMs: number = DEFAULT_RETRY_DELAY_MS
-): Promise<axios.AxiosResponse> => {
-  let response: axios.AxiosResponse;
-  const ax = new axios.Axios();
-
-  do {
-    response = await ax.request({ url, ...requestParams });
-    if (response.status < HttpStatusCode.InternalServerError) {
-      return response;
-    }
-
+): Promise<AxiosResponse | never> => {
+  const maxRetries = numRetries;
+  const requestConfig = { url, ...requestParams };
+
+  // Without the following initial request before the for loop, TS complains:
+  // "Variable 'response' is used before being assigned"
+  let response: AxiosResponse = await axios.request(requestConfig);
```

Comment:
If this initial request throws an exception (network error, timeout, etc.), the for loop will never execute, but the exception will propagate up.

Recommended fix:
```typescript
let response: AxiosResponse;
let attemptsLeft = numRetries + 1; // +1 for initial attempt

do {
  response = await axios.request(requestConfig);
  
  if (response.status < HttpStatusCode.InternalServerError) {
    return response;
  }
  
  if (--attemptsLeft > 0) {
    const backedoffDelayMs = retryDelayMs * Math.pow(2, numRetries - attemptsLeft);
    await delay(backedoffDelayMs);
  }
} while (attemptsLeft > 0);

return response;
```

Reply:
do...while doesn't respect async...await. Hence, the implementation was updated to run with a for loop.